ocr_service/api/endpoints/extraction.py

```
@@ -0,0 +1,294 @@
+                content_length = int(head_response.headers.get('content-length', 0))
+                
+                if content_length > LARGE_FILE_THRESHOLD:
+                    logger.info(f"Warning: Large file detected ({content_length/1_000_000:.2f}MB). Processing may take longer.")
+                    
+            except Exception as e:
+                logger.warning(f"Could not get content length: {e}")
+            
+            # Download the file from the provided URL
+            response = requests.get(processed_url, headers=self.headers, timeout=REQUEST_TIMEOUT)
+            if response.status_code != 200:
+                raise Exception(f"Failed to fetch file from URL: HTTP {response.status_code}")
```

Ruff (0.8.2)

Abstract `raise` to an inner function

(TRY301)